lib/kql/kql/parser.py

```diff
@@ -376,7 +377,14 @@ def lark_parse(text):
     walker = BaseKqlParser(text)
 
     try:
-        return lark_parser.parse(text)
+        tree = lark_parser.parse(text)
+
+        # Check for whitespace around "and" and "or" tokens
+        lines = text.split('\n')
+        check_whitespace(collect_token_positions(tree, "and"), 'and', lines)
+        check_whitespace(collect_token_positions(tree, "or"), 'or', lines)
```

Comment:
it might be more optimized to do a single traversal to collect all the token values (instead of traversing twice)

Reply:
Agreed, it would save the second (or potentially third) traversal, no particular need to parse twice as far as I can recall.